### PR TITLE
8325591: [Mac] DRAG_DONE reports null transferMode when destination is external

### PR DESCRIPTION
**Pull request by beldenfox**
At the end of a drag operation the Mac Glass code sends out a DRAG_DONE event using the operation mask tracked in the GlassDragSource to determine the final transfer mode. That mask is only updated when a window in the JavaFX app is the drop destination. If the drag moves to an external destination the mask is set to NONE. If the drag terminates in the external destination that NONE forms the basis of the transfer mode sent via the DRAG_DONE event.

At the very end of the drag the OS calls the NSDraggingSource (GlassDraggingSource) with the final drag operation. This PR issues the DRAG_DONE from that callback so it can get the final transfer mode correct for both internal and external destinations.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8325591](https://bugs.openjdk.org/browse/JDK-8325591): [Mac] DRAG_DONE reports null transferMode when destination is external (**Bug** - P4)

### Reviewers
 * [Lukasz Kostyra](https://openjdk.org/census#lkostyra) (@lukostyra - Committer)
 * [Kevin Rushforth](https://openjdk.org/census#kcr) (@kevinrushforth - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1371/head:pull/1371` \
`$ git checkout pull/1371`

Update a local copy of the PR: \
`$ git checkout pull/1371` \
`$ git pull https://git.openjdk.org/jfx.git pull/1371/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1371`

View PR using the GUI difftool: \
`$ git pr show -t 1371`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1371.diff">https://git.openjdk.org/jfx/pull/1371.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1371#issuecomment-1949435955)

### CODE REVIEW

**Comment by kevinrushforth** [comment]
Reviewers: @kevinrushforth @lukostyra

**Review by lukostyra — APPROVED** [review]
LGTM

**Review by kevinrushforth — COMMENTED** [review]
The code changes look good. I did a quick test and it seems fine.

Would you be able to provide a manual test program for this case, under `tests/manual/dnd`? Also, can you merge in the latest changes from `master`?

File: tests/manual/dnd/DndBasic.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+            result.setText(message);
+            evt.consume();
+        });
+        source.setFill(Color.LIGHTGREEN);
+
+        return new Group(source, labels);
+    }
+
+    private Group createTarget(TransferMode[] modes)
+    {
```
Comment (kevinrushforth) [inline tests/manual/dnd/DndBasic.java:169]: Minor: move brace to previous line

**Review by kevinrushforth — APPROVED** [review]
Looks good. Thanks for the new manual test program. I left one minor formatting comment on the test, but I'll approve it anyway (and reapprove if you update it).

**Review by kevinrushforth — APPROVED** [review]

**Comment by kevinrushforth** [comment]
@lukostyra Can you please re-review?

**Review by lukostyra — APPROVED** [review]
LGTM

**Pull request closed without merging**